consul/server.go

```diff
@@ -327,7 +330,7 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string, w
 // setupRaft is used to setup and initialize Raft
 func (s *Server) setupRaft() error {
 	// If we are in bootstrap mode, enable a single node cluster
-	if s.config.Bootstrap {
+	if s.config.Bootstrap || s.config.DevMode {
```

Comment:
The Raft config is only available from a `*consul.Config` and not an `*agent.Config` so we have to check on the Consul side when we set the `EnableSingleNode` flag.